pkg/apis/logging/v1/kibana_types.go

```diff
@@ -0,0 +1,60 @@
+package v1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KibanaSpec defines the desired state of Kibana
+// +k8s:openapi-gen=true
+type KibanaSpec struct {
+	Image           string                   `json:"image"`
+	ManagementState ManagementState          `json:"managementState"`
+	Resources       *v1.ResourceRequirements `json:"resources"`
+	NodeSelector    map[string]string        `json:"nodeSelector,omitempty"`
+	Tolerations     []v1.Toleration          `json:"tolerations,omitempty"`
+	Replicas        int32                    `json:"replicas"`
+	ProxySpec       `json:"proxy,omitempty"`
```

Comment:
Should this become a pointer so users can choose not to define it?

Reply:
we don't need to do that in order to not define it. the only difference is as a pointer this field could be null, otherwise its just empty